iceoryx2-ffi/cxx/include/iox2/config.hpp

```diff
@@ -13,10 +13,229 @@
 #ifndef IOX2_CONFIG_HPP
 #define IOX2_CONFIG_HPP
 
+#include "iox/duration.hpp"
+#include "iox/file_name.hpp"
+#include "iox/path.hpp"
+#include "iox2/config_creation_error.hpp"
 #include "iox2/internal/iceoryx2.hpp"
+#include "iox2/unable_to_deliver_strategy.hpp"
 
 namespace iox2 {
 class Config;
+
+namespace config {
+class Global;
+
+/// All configurable settings of a [`Node`].
+class Node {
+  public:
+    /// The directory in which all node files are stored
+    auto directory() && -> const char*;
+    /// Set the directory in which all node files are stored
+    void set_directory(const iox::Path& value) &&;
+    /// The suffix of the monitor token
+    auto monitor_suffix() && -> const char*;
+    /// Set the suffix of the monitor token
+    void set_monitor_suffix(const iox::FileName& value) &&;
+    /// The suffix of the files where the node configuration is stored.
+    auto static_config_suffix() && -> const char*;
+    /// Set the suffix of the files where the node configuration is stored.
+    void set_static_config_suffix(const iox::FileName& value) &&;
+    /// The suffix of the service tags.
+    auto service_tag_suffix() && -> const char*;
+    /// Set the suffix of the service tags.
+    void set_service_tag_suffix(const iox::FileName& value) &&;
+    /// When true, the [`NodeBuilder`](NodeBuilder) checks for dead nodes and
+    /// cleans up all their stale resources whenever a new [`Node`](Node) is
+    /// created.
+    auto cleanup_dead_nodes_on_creation() && -> bool;
+    /// Enable/disable the cleanup dead nodes on creation
+    void set_cleanup_dead_nodes_on_creation(bool value) &&;
+    /// When true, the [`NodeBuilder`](NodeBuilder) checks for dead nodes and
+    /// cleans up all their stale resources whenever an existing [`Node`](Node) is
+    /// going out of scope.
+    auto cleanup_dead_nodes_on_destruction() && -> bool;
+    /// Enable/disable the cleanup dead nodes on destruction
+    void set_cleanup_dead_nodes_on_destruction(bool value) &&;
+
+  private:
+    friend class Global;
+    explicit Node(iox2_config_h* config);
+
+    iox2_config_h* m_config;
+};
+
+/// All configurable settings of a [`Service`].
+class Service {
+  public:
+    /// The directory in which all service files are stored
+    auto directory() && -> const char*;
+    /// Set the directory in which all service files are stored
+    void set_directory(const iox::Path& value) &&;
+    /// The suffix of the publishers data segment
+    auto publisher_data_segment_suffix() && -> const char*;
```

Comment:
To make this better certifiable, we might need to introduce something like `iox::Str`